runtime/vm/ContinuationHelpers.cpp

```diff
@@ -141,12 +161,11 @@ yieldContinuation(J9VMThread *currentThread)
 {
 	BOOLEAN result = TRUE;
 	J9VMContinuation *continuation = currentThread->currentContinuation;
-
 	Assert_VM_notNull(currentThread->currentContinuation);
 
-	VM_ContinuationHelpers::swapFieldsWithContinuation(currentThread, continuation);
-	continuation->carrierThread = NULL;
 	currentThread->currentContinuation = NULL;
+	VM_ContinuationHelpers::swapFieldsWithContinuation(currentThread, continuation);
+	continuation->carrierThreadAndConcurrentlyScanned = 0;
```

Comment:
We need a write fence for weakly order CPUs just before the last step when we announce to the world that unmounting is complete (via resetting sync variable carrierThreadAndConcurrentlyScanned), to make sure rest of CPUs see previous updates like 'swap' complete. 

But there is already a full fence just after, so we could move carrierThreadAndConcurrentlyScanned reset the last. @gacholio, @babsingh is that ok (not sure who added this fence and what is real meaning of it)?

Reply:
This will require a bit of thought - write barriers in general serve two purposes - to be matched with read barriers to avoid heavyweight synchronization, or to present subtructures properly initialized:
```
initialize array X
writeBarrier();
store X pointer into structure Y
present Y to the world
```
There are instances of this pattern already in the VM:
https://github.com/eclipse-openj9/openj9/blob/78db7ef045da31bdbe2f0032d6544f4346e8dbb4/runtime/vm/jnicsup.cpp#L2133-L2167

Reply:
We need it for the latter reason. Resetting carrierThreadAndConcurrentlyScanned effectively presents the continuation structure to GC and letting it concurrently scan. But GC pottentially running on another CPU must see the up-to-date continuation structure that just have been extensively update by 'swap'

I don't fully understand the former reason (but I guess it has something to do with lock-free inNative?)...